SharedCode/src/commonMain/kotlin/common.kt

```diff
@@ -68,26 +68,44 @@ fun toHumanReadableDate(epochMillis: Long, dateInfo: DateFormatInfo): String {
     // If now.day == date.day || diff < 12h
     if (now.dayOfYear == date.dayOfYear || diff.hours < 12) {
         // Display as HH:mm (in xx h/m)
-        val diffStr = if (diff.hours < 1) "${diff.minutes.toInt()} min" else "${diff.hours.toInt()} hr"
+        val diffStr = formatDiff(dateInfo.now, epochMillis)
         return "${hourFormatter.format(dateOffset)} (in $diffStr)"
     }
     return dateFormatter.format(dateOffset)
 }
 
+fun formatDiff(date1Millis: Long, date2Millis: Long): String {
+    val date1 = DateTime.fromUnix(date1Millis)
+    val date2 = DateTime.fromUnix(date2Millis)
+    val diff = date2 - date1
+    if (date1.dayOfYear == date2.dayOfYear || diff.hours < 12) {
+        if (diff.hours < 1) {
+            return "${diff.minutes.toInt()} min"
+        } else {
+            return "${diff.hours.toInt()}:${"${(diff.minutes.toInt() % 60)}".padStart(2, '0')}"
```

Comment:
I'm not sure that giving the precise difference is better than just saying 'in 2 hours'. It looks a _little_ cluttered.

Reply:
That's fair, I was just wondering for train duration you probably want minutes as well. Maybe I'll add a flag.

Reply:
Added the flag